### ISSUE TITLE
`parse-url`  should take an optional parameter that controls error handling

### ISSUE THREAD

**Issue by mrange** [schibsted/jslt#152]
Hi.

`parse-url` throws on failure. This is great for many cases but we receive data where we don't know before-hand if it's a valid URL or not. Adding a pre-cleanup step using regex is cumbersome and probably never will catch all problems.

The proposal is that instead allow `parse-url` error handling to be controlled by the controller by introducing an optional parameter `throwOnFailure`. 

If the parameter isn't set or set to `true` the current behavior is kept to ensure backwards compatibility.

If the parameter is set `parse-url` returns a failure object for example like this:

```
parse-url("this-is-an-invalid-url", false)
```

Yields:

```json
      {
        "error": "MalformedURLException",
        "message": "no protocol: this-is-an-invalid-url",
        "input": "this-is-an-invalid-url"
      }
```

Code that wish to do branch on whether there's a failure or not can test if `error` exists:

```
let x = parse-url("this-is-an-invalid-url", false)
if ($x.error) do-error-handling() else do-normal-ops()
```

**Comment by larsga** [comment]
You're right. This was sloppy of us. We definitely need error handling here.

The pattern we've used elsewhere is to have a second fallback parameter that gets returned if the value is invalid. In this case that means losing information about how the URL was invalid. Which may or may not be important.

Your suggestion is not in line with how the other functions work, but preserves potentially useful error information. So maybe that's the right choice.

Hmmm.

**Comment by mrange** [comment]
I have a PR in the works. Hopefully in a few mins so you can see how I thought it could be fixed.
